src/nni_manager/common/utils.ts

```diff
@@ -363,6 +374,103 @@ async function getVersion(): Promise<string> {
     return deferred.promise;
 } 
 
+function getTunerProc(command: string, stdio: StdioOptions, newCwd: string, newEnv: any):ChildProcess{
+    let cmd: string = command;
+    let arg: string[] = [];
+    let newShell: boolean = true;
+    if(process.platform === "win32"){
+        command = command.split('python3').join('python');
+        cmd = command.split(" ", 1)[0];
+        arg = command.substr(cmd.length+1).split(" ");
+        newShell = false;
+    }
+    const tunerProc: ChildProcess = spawn(cmd, arg, {
+        stdio,
+        cwd: newCwd,
+        env: newEnv,
+        shell: newShell
+    });
+    return tunerProc;
+}
+
+async function isAlive(pid:any):Promise<boolean>{
+    let deferred : Deferred<boolean> = new Deferred<boolean>();
+    let alive: boolean = false;
+    if(process.platform ==='win32'){
+        const str = cp.execSync(`tasklist /FI ${'"PID eq '+pid +'"'}`).toString();
+        if(!str.includes("No tasks")){
+            alive = true;
+        }
+    }
+    else{
+        try {
+            await cpp.exec(`kill -0 ${pid}`);
+            alive = true;
+        } catch (error) {
+            //ignore
+        }
+    }
+    deferred.resolve(alive);
+    return deferred.promise;
+}
+
+async function killPid(pid:any):Promise<void>{
+    let deferred : Deferred<void> = new Deferred<void>();
+    try {
+        if (process.platform === "win32") {
+            await cpp.exec(`powershell.exe kill ${pid}`);
+        }
+        else{
+            await cpp.exec(`kill ${pid}`);
+        }
+    } catch (error) {
+        // pid does not exist, do nothing here
+    }
+    deferred.resolve();
+    return deferred.promise;
+}
+
+async function runScript(localTrailConfig: TrialConfig, workingDirectory: string, variables: { key: string; value: string }[]):Promise<string[]>{
```

Comment:
Suggest to move this runScript function to somewhere in local training service folder. It is not a common util method.